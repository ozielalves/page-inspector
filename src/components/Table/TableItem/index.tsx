import React from "react";
import styled from "styled-components";
import { TableItemContent, Text, Wrapper, TableCell } from "../../../styles";

interface TableItemProps {
  id: string;
  keyword: string;
  status: string;
  setFiler: React.Dispatch<React.SetStateAction<string>>;
}

const Table = ({ keyword, status, setFiler, id }: TableItemProps) => {
  return (
    <StyledWrapper onClick={() => setFiler(id)}>
      <ClickableTableItem
        style={{
          backgroundColor:
            status === "done"
              ? "var(--base-color-green)"
              : "var(--base-color-deep-orange)",
        }}
      >
        <TableCell>
          <StyledText>{keyword}</StyledText>
        </TableCell>
        <TableCell style={{ width: 70, justifyContent: "center" }}>
          <StyledText>{status}</StyledText>
        </TableCell>
      </ClickableTableItem>
    </StyledWrapper>
  );
};

const ClickableTableItem = styled(TableItemContent)`
  cursor: pointer;
  transition: opacity 0.5s ease;

  &:hover {
    opacity: 0.7;
  }
`;

const StyledText = styled(Text)`
  font-weight: bold;
  font-size: 18px;
  color: var(--base-color-white);
`;

const StyledWrapper = styled(Wrapper)`
  flex-direction: column;
  align-items: flex-start;
  overflow-y: auto;
  position: relative;
  padding: 0 10px 17px 10px;

  @media (max-width: 600px) {
    padding: 0 0 17px 0;
  }
`;

export default Table;
