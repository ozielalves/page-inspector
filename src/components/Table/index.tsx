import React from "react";
import { Request } from "../../db/repositories/requests";
import styled from "styled-components";
import TableItem from "./TableItem";
import { TableItemContent, Text, Wrapper, TableCell } from "../../styles";
import CircularProgress from "../CircularProgress";

interface TableProps {
  data?: Request[];
  setFilter: React.Dispatch<React.SetStateAction<string>>;
  handleDelete: (id: string) => void;
}

const Table = ({ data, setFilter, handleDelete }: TableProps) => {
  return (
    <StyledWrapper>
      <StyledTableItem>
        <TableCell>
          <StyledText style={{ whiteSpace: "normal" }}>
            Palavra Chave
          </StyledText>
        </TableCell>
        {/* <div> */}
        <TableCell style={{ justifyContent: "flex-end" }}>
          <StyledText>Status</StyledText>
        </TableCell>
        <FakeCell />
        {/* </div> */}
      </StyledTableItem>
      <RowWrapper>
        {data ? (
          data.map(({ id, keyword, status }: Request) => (
            <TableItem
              key={id}
              id={id!}
              keyword={keyword!}
              status={status}
              setFilter={setFilter}
              handleDelete={handleDelete}
            />
          ))
        ) : (
          <CircularProgress size={100} color="grey" />
        )}
      </RowWrapper>
    </StyledWrapper>
  );
};

const FakeCell = styled.div`
  width: 1px;
  margin: 20px;

  @media (max-width: 600px) {
    margin: 10px;
  }
`;

const StyledTableItem = styled(TableItemContent)`
  margin: 10px 40px 17px 40px;

  @media (max-width: 600px) {
    margin: 10px 36px;
  }
`;

const StyledText = styled(Text)`
  font-weight: bold;
  font-size: 18px;
  color: var(--base-color-grey);
`;

const StyledWrapper = styled(Wrapper)`
  flex-direction: column;
  justify-content: flex-start;
  height: 90%;
  width: 75%;
  padding: 0 10px;

  @media (max-width: 1024px) {
    width: 100%;
    padding: 0;
  }
`;

const RowWrapper = styled(StyledWrapper)`
  width: 100%;
  overflow-x: hidden;
  overflow-y: auto;
  padding: 0;
  height: 100%;
  border-radius: 25px;
  padding-right: 20px;

  &::-webkit-scrollbar {
    width: 4px;
  }

  &::-webkit-scrollbar-track {
    box-shadow: inset 0 0 5px transparent;
    border-radius: 10px;
  }

  &::-webkit-scrollbar-thumb {
    background: #8c8c8c;
    border-radius: 10px;
  }

  @media (max-width: 1024px) {
    padding: 0;
    padding-right: 10px;
  }
`;

export default Table;
