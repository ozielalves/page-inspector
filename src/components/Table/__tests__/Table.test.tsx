import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import React from "react";
import Table from "../index";

import TableItem from "../TableItem";

describe("Table render test", () => {
  const data = [
    {
      id: "OJOaad",
      apiId: "POossSE",
      status: "done",
      urls: ["www.exemplo.com", "www.exemplo.com", "www.exemplo.com"],
    },
    {
      id: "sdfsdf",
      apiId: "sdfwewr",
      status: "active",
      urls: ["www.exemplo.com", "www.exemplo.com", "www.exemplo.com"],
    },
  ];

  it("renders correctly ActionButtom", () => {
    const wrapper = shallow(
      <Table data={data} handleDelete={() => {}} setFilter={() => {}} />
    );
    expect(wrapper.find(TableItem)).toBeTruthy();
  });

  it("should render the correct Table", () => {
    const tree = renderer
      .create(
        <Table data={data} handleDelete={() => {}} setFilter={() => {}} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
