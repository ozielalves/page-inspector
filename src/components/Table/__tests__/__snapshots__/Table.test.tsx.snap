// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table render test should render the correct Table 1`] = `
<div
  className="sc-gKsewC sc-fodVxV eygLjf gyzLBv"
>
  <div
    className="sc-jSgupP sc-pFZIQ sc-kstrdz gmVoiw eoaBsr jsLnef"
  >
    <div
      className="sc-gKsewC sc-jrAGrp eygLjf jIqWmM"
    >
      <p
        className="sc-gsTCUz sc-hBEYos hFbaQT eOjMou"
        style={
          Object {
            "whiteSpace": "normal",
          }
        }
      >
        Palavra Chave
      </p>
    </div>
    <div
      className="sc-gKsewC sc-jrAGrp eygLjf jIqWmM"
      style={
        Object {
          "justifyContent": "flex-end",
        }
      }
    >
      <p
        className="sc-gsTCUz sc-hBEYos hFbaQT eOjMou"
      >
        Status
      </p>
    </div>
    <div
      className="sc-bqyKva fAChft"
    />
  </div>
  <div
    className="sc-gKsewC sc-fodVxV sc-fFubgz eygLjf gyzLBv bEOkRe"
  >
    <div
      className="sc-jSgupP sc-pFZIQ sc-crrsfI gmVoiw eoaBsr jfeaQS"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "var(--base-color-green)",
        }
      }
    >
      <div />
      <div
        className="sc-gKsewC sc-jrAGrp eygLjf jIqWmM"
      >
        <p
          className="sc-gsTCUz sc-dQppl hFbaQT isdmtl"
        />
      </div>
      <div
        className="sc-gKsewC sc-jrAGrp eygLjf jIqWmM"
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <p
          className="sc-gsTCUz sc-dQppl hFbaQT isdmtl"
        >
          done
        </p>
        <div
          className="sc-kEjbxe jqePyN"
          style={
            Object {
              "width": "30%",
            }
          }
        >
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "var(--base-color-error)",
              }
            }
          >
            <svg>
              trash-icon.svg
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      className="sc-jSgupP sc-pFZIQ sc-crrsfI gmVoiw eoaBsr jfeaQS"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "var(--base-color-deep-orange)",
        }
      }
    >
      <div />
      <div
        className="sc-gKsewC sc-jrAGrp eygLjf jIqWmM"
      >
        <p
          className="sc-gsTCUz sc-dQppl hFbaQT isdmtl"
        />
      </div>
      <div
        className="sc-gKsewC sc-jrAGrp eygLjf jIqWmM"
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <progress
          className="sc-iqHYGH emUSPH"
          style={
            Object {
              "color": "var(--base-color-white)",
              "height": 30,
              "width": 30,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
